proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/ProxyNodeLogger.java

```diff
@@ -15,6 +16,15 @@ public void addContext(ProxyNodeMDCKey key, String value) {
         MDC.put(key.name(), value);
     }
 
+    public void addContext(Exception exception) {
+        final String cause = Optional.ofNullable(exception.getCause()).map(Throwable::getMessage).orElse(null);
```

Comment:
Can we make this easier to read by replacing with:
`Throwable cause = exception.getCause();
        if (cause != null) {
            addContext(ProxyNodeMDCKey.EXCEPTION_CAUSE, cause.getMessage());
        }`

Reply:
Done.